test/spec/EnterSpec.hs

```diff
@@ -215,3 +215,81 @@ spec = do
             stdout output `shouldStartWith` "v18."
             output <- runGarn ["enter", "frontend"] "npm --version\nexit\n" repoDir Nothing
             stdout output `shouldStartWith` "9."
+
+        describe "top-level environments" $ around onTestFailureLogger $ do
+          it "allows entering top-level environments" $ \onTestFailureLog -> do
+            writeFile
+              "garn.ts"
+              [i|
+                import * as garn from "#{repoDir}/ts/mod.ts"
+
+                const testTool = garn.build(`
+                  mkdir -p $out/bin
+                  echo 'echo test tool output' > $out/bin/test-tool
+                  chmod +x $out/bin/test-tool
+                `);
+
+                export const topLevelEnv = garn.emptyEnvironment.withDevTools([testTool]);
+              |]
+            output <- runGarn ["enter", "topLevelEnv"] "test-tool\nexit\n" repoDir Nothing
+            onTestFailureLog output
+            stdout output `shouldBe` "test tool output\n"
+
+          it "does not show a description for top-level environments" $ \onTestFailureLog -> do
+            writeFile
+              "garn.ts"
+              [i|
+                import * as garn from "#{repoDir}/ts/mod.ts"
+
+                export const topLevelEnv = garn.emptyEnvironment;
+              |]
+            output <- runGarn ["enter", "--help"] "" repoDir Nothing
+            onTestFailureLog output
+            stdout output
+              `shouldMatch` unindent
+                [i|
+                  Available commands:
+                    topLevelEnv[ ]*
```

Comment:
Wouldn't this match even things that have description? Like maybe a `$` is needed at the end?

Reply:
Turns out both works as intended. But with the `$` it's more obvious, so I'll change it.